src/Microsoft.AspNetCore.Razor.Design/Internal/AssemblyTagHelperDescriptorResolver.cs

```diff
@@ -61,6 +52,47 @@ public IEnumerable<TagHelperDescriptor> Resolve(string assemblyName, ErrorSink e
                         DesignResources.InvalidProtocolValue,
                         typeof(TagHelperDescriptor).FullName, ProtocolVersion));
             }
+
+            if (!_loadedAssembly)
+            {
+                LoadMvcAssembly();
+            }
+
+            if (_hasMvcAssembly)
+            {
+                var serviceCollection = new ServiceCollection();
+                serviceCollection.AddSingleton<ITagHelperTypeResolver>(_tagHelperTypeResolver);
+                serviceCollection.AddSingleton<ITagHelperDescriptorFactory>(_tagHelperDescriptorFactory);
+
+                // Populate the service collection
+                _populateMethod.Invoke(_serviceCollectionProviderClassInstance, new object[] { serviceCollection, assemblyName });
+
+                var services = serviceCollection.BuildServiceProvider();
+                var tagHelperDescriptorResolver = services.GetRequiredService<ITagHelperDescriptorResolver>();
+
+                var directiveDescriptors = new TagHelperDirectiveDescriptor[]
```

Comment:
You could share the logic from here on down with the falling back to manually resolving descriptors.

Aka, if MVC isn't in the project:

```C#
TagHelperDescriptorResolver tagHelperDescriptorResolver;
if (_hasMvcAssembly)
{
     // Do work to get tagHelperDescriptorResolver from MVC
}
else
{
    tagHelperDescriptorResolver = new TagHelperDescriptorResolver(_tagHelperTypeResolver, _tagHelperDescriptorFactory);
}

// Build resolution context and resolve
// Win
```